cmd/ddev/cmd/dev_sequelpro.go

```diff
@@ -6,69 +6,108 @@ import (
+	app, err := getActiveApp()
+	if err != nil {
+		return "", err
+	}
 
-		if !dockerutil.IsRunning(nameContainer) {
-			util.Failed("App not running locally. Try `ddev start`.")
-		}
+	if len(args) != 0 {
+		return "", fmt.Errorf("invalid arguments to sequelpro command: %v", args)
+	}
+	nameContainer := fmt.Sprintf("%s-db", app.ContainerName())
 
-		mysqlContainer, err := dockerutil.GetContainer(nameContainer)
-		if err != nil {
-			log.Fatal(err)
-		}
+	if !dockerutil.IsRunning(nameContainer) {
+		return "", errors.New("app not running locally. Try `ddev start`")
+	}
 
-		dbPort, err := dockerutil.GetDockerPublicPort(mysqlContainer, int64(3306))
-		if err != nil {
-			log.Fatal(err)
-		}
+	mysqlContainer, err := dockerutil.GetContainer(nameContainer)
+	if err != nil {
+		return "", err
+	}
 
-		tmpFilePath := path.Join(app.AppRoot(), "sequelpro.spf")
-		tmpFile, err := os.Create(tmpFilePath)
-		if err != nil {
-			log.Fatalln(err)
-		}
-		defer util.CheckClose(tmpFile)
-
-		_, err = tmpFile.WriteString(fmt.Sprintf(
-			platform.SequelproTemplate,
-			"data",                  //dbname
-			"127.0.0.1",             //host
-			mysqlContainer.Names[0], //container name
-			"root",                  // dbpass
-			strconv.FormatInt(dbPort, 10), // port
-			"root", //dbuser
-		))
-		util.CheckErr(err)
-
-		err = exec.Command("open", tmpFilePath).Run()
-		if err != nil {
-			log.Fatal(err)
-		}
+	dbPort := appports.GetPort("db")
+	if err != nil {
+		return "", err
+	}
+
+	tmpFilePath := path.Join(app.AppRoot(), ".ddev/sequelpro.spf")
+	tmpFile, err := os.Create(tmpFilePath)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	defer util.CheckClose(tmpFile)
+
+	_, err = tmpFile.WriteString(fmt.Sprintf(
+		platform.SequelproTemplate,
+		"data",                  //dbname
+		"127.0.0.1",             //host
+		mysqlContainer.Names[0], //container name
+		"root",                  // dbpass
+		dbPort,                  // port
+		"root",                  //dbuser
+	))
+	util.CheckErr(err)
+
+	err = exec.Command("open", tmpFilePath).Run()
+	if err != nil {
+		return "", err
+	}
+	return "sequelpro command finished successfully!", nil
+}
 
-		util.Success("sequelpro command finished successfully!")
+// dummyDevSequelproCmd represents the "not available" sequelpro command
+var dummyDevSequelproCmd = &cobra.Command{
+	Use:   "sequelpro",
+	Short: "This command is not available since sequel pro.app is not installed",
+	Long:  `Where installed, "ddev sequelpro" launches the sequel pro database browser`,
+	Run: func(cmd *cobra.Command, args []string) {
+		util.Failed("The sequelpro command is not available because sequel pro.app is not detected on your workstation")
 
 	},
 }
 
+// init installs the real command if it's available, otherwise dummy command
 func init() {
-	RootCmd.AddCommand(LocalDevSequelproCmd)
+	if detectSequelpro() {
+		RootCmd.AddCommand(localDevSequelproCmd)
+	} else {
+		RootCmd.AddCommand(dummyDevSequelproCmd)
```

Comment:
Perhaps there are two different use cases here:

- It'd be nice if it was discoverable for OSX users, so they know they can take advantage of it if they install sequelpro
- it feels like noise for linux/windows users, where there is no hope of gaining this functionality.